File: src/ObjectManager/Object.Tes/FilePacks/BsaFile+Extensions.cs
using System;
using System.Linq;

namespace OA.Tes.FilePacks
{
    partial class BsaFile
    {
        void TestContainsFile()
        {
            foreach (var file in _files)
            {
                Console.WriteLine(file.Path);
                if (!ContainsFile(file.Path))
                    throw new FormatException("Hash Invalid");
                else if (!_filesByHash[HashFilePath(file.Path)].Any(x => x.Path == file.Path))
                    throw new FormatException("Hash Invalid");
            }
        }

        void TestLoadFileData()
        {
            foreach (var file in _files)
            {
                Console.WriteLine(file.Path);
                LoadFileData(file);
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/BsaFile+Extensions.swift
//
//  BsaFile+Extensions.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

import Foundation

extension BsaFile {
    func testContainsFile() {
        for file in _files {
            debugPrint(file.path, file.pathHash)
            guard containsFile(file.path) else {
                fatalError("Hash Invalid")
            }
            guard _filesByHash[hashFilePath(file.path)]!.contains(where: { $0.path == file.path }) else {
                fatalError("Hash Invalid")
            }
        }
    }

    func testLoadFileData() {
        for file in _files {
            debugPrint(file.path)
            _ = loadFileData(file: file)
        }
    }
}
